### DISCUSSION TITLE
How can I clear 1px width blank in Grid?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13049] [Q&A] [answered]
Here is my XAML:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.MainWindow"
        Title="AvaloniaApplication1">
	<Grid RowDefinitions="auto,auto">
		<Grid ColumnDefinitions="auto,auto,auto">
			<Border Height="80" Width="80" Background="Red"></Border>
			<Border Height="80" Width="150" BorderBrush="Red" BorderThickness="0,5,0,5" Grid.Column="1"></Border>
			<Border Height="80" Width="80" Background="Red" Grid.Column="2"></Border>
		</Grid>
		<StackPanel Orientation="Horizontal" Grid.Row="1" Margin="0,30,0,0">
			<Border Height="80" Width="80" Background="Red"></Border>
			<Border Height="80" Width="150" BorderBrush="Red" BorderThickness="0,5,0,5" Grid.Column="1"></Border>
			<Border Height="80" Width="80" Background="Red" Grid.Column="2"></Border>
		</StackPanel>
	</Grid>
</Window>
```

And here is the result:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/9124d75e-16e2-4b97-91e9-7d6d8b9c2515)

As you see, there is 1px width between the first and the second Border in Grid. What's more, I notice that this problem only occurs in Grid while StackPanel won't like this.

The version I am using is 11.0.4

How can I solve this?

**Comment by thevortexcloud** [comment]
There may be a default margin somewhere. You can try clearing it by setting it explicitly. EG `margin="0"`

**Reply by kyuranger** [reply]
It doesn't work any.

**Comment by timunie** [comment]
Check DevTools [F12] and inspect visual and logical tree

**Reply by kyuranger** [reply]
Both visual tree and logical tree are fine. The Width in the DesizreSize attribute in the Grid is also 310px, which is correct. I can't find the problem

**Comment by rabbitism** [comment, marked as answer]
You can either set UseLayoutRounding=False or HorizontalAlignment=Left on any one of your Grid.

**Reply by rabbitism** [reply]
This is only a work around. The calculation is weird but I haven't get a chance to look at it.

**Reply by kyuranger** [reply]
Thank you. I have tested that both your ways work. I heard that Avalonia is pixel-perfect, but I didn’t expect that it also has flaws.

**Reply by timunie** [reply]
rounding issues :-(

**Reply by kyuranger** [reply]
Thank you all the same.
